druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/entity/CacheActionRunner.java

```diff
@@ -15,29 +12,39 @@
 import com.yahoo.maha.maha_druid_lookups.server.lookup.namespace.MonitoringConstants;
 import com.yahoo.maha.maha_druid_lookups.server.lookup.namespace.RocksDBManager;
 import com.metamx.common.logger.Logger;
+import com.yahoo.maha.maha_druid_lookups.server.lookup.namespace.schema.BaseSchemaFactory;
+import com.yahoo.maha.maha_druid_lookups.server.lookup.namespace.schema.protobuf.ProtobufSchemaFactory;
 import org.rocksdb.RocksDB;
 
-import java.io.Serializable;
-import java.util.Objects;
 import java.util.Optional;
 
-public class CacheActionRunner {
+public class CacheActionRunner implements BaseCacheActionRunnerChain {
 
     private static final Logger LOG = new Logger(CacheActionRunner.class);
 
+    private BaseCacheActionRunnerChain nextChain = new CacheActionRunnerFlatBuffer();
+
+    @Override
     public byte[] getCacheValue(final String key
             , Optional<String> valueColumn
             , final Optional<DecodeConfig> decodeConfigOptional
             , RocksDBManager rocksDBManager
-            , ProtobufSchemaFactory protobufSchemaFactory
+            , BaseSchemaFactory schemaFactory
             , LookupService lookupService
             , ServiceEmitter emitter
-            , RocksDBExtractionNamespace extractionNamespace){
+            , RocksDBExtractionNamespace extractionNamespace) {
+
         try {
             if (!extractionNamespace.isCacheEnabled() && valueColumn.isPresent()) {
                 return lookupService.lookup(new LookupService.LookupData(extractionNamespace, key, valueColumn.get(), decodeConfigOptional));
             }
 
+            if (routeToNextChain(schemaFactory)) {
```

Comment:
Not sure if chain of responsibility is the right pattern here, since there's no 'chaining' of steps. 'schemaFactory' object can be type flatbuffer, protobuf or some other derived class of BaseSchemaFactory. Falling back to protobuf silently may not be the expected behavior always.

Reply:
I second that. there is no chain of responsibility here. you should call appropriate CacheActionRunner based on the configuration in the ExtractionNamespace which is already done here https://github.com/yahoo/maha/blob/master/druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/query/lookup/namespace/RocksDBExtractionNamespace.java#L80-L90

Reply:
removed the chain of responsibility.

Reply:
and added new cache action runner.